Test failures on aarch64-darwin

### Description

I'm seeing the following test failures when testing on aarch64-darwin:
```
=========================== short test summary info ============================
FAILED tests/lax_control_flow_test.py::LaxControlFlowTest::testScanGrad_jit_scan=False_jit_f=False_impl=unroll2 - AssertionError: 
FAILED tests/for_loop_test.py::ForLoopTransformationTest::test_for_loop_fixpoint_correctly_identifies_loop_varying_residuals_unrolled_for_loop - AssertionError: 
FAILED tests/qdwh_test.py::QdwhTest::testQdwhWithRandomMatrix3 - AssertionError: 
========== 3 failed, 11328 passed, 1205 skipped in 513.09s (0:08:33) ===========
```
In particular, I'm running on an Apple M1 Max 2021 chip. Complete test logs [can be seen here](https://gist.github.com/samuela/fe8420c2635abbabfe955deb9997db02). All the test failures appear to be numerical precision issues. It looks like tolerances just need to be made slightly more generous.

I'm running with the following pytest flags:
```nix
  pytestFlagsArray = [
    "--numprocesses=4"
    "-W ignore::DeprecationWarning"
    "tests/"
  ];
```

### What jax/jaxlib version are you using?

jax v0.4.5, jaxlib v0.4.4

### Which accelerator(s) are you using?

n/a

### Additional system info

Python python3-3.10.10 from Nixpkgs

### NVIDIA GPU info

n/a

--- a/tests/lax_control_flow_test.py
+++ b/tests/lax_control_flow_test.py
@@ -1662,6 +1662,16 @@ class LaxControlFlowTest(jtu.JaxTestCase):
       assert b.shape == ()
       return c, b
 
+    if scan is scan_with_new_checkpoint:
+      rtol = {np.float32: 5e-5, np.float64: 1e-13}
+      atol = 1e-5
+    elif scan is scan_with_for:
+      rtol = {np.float32: 2e-5, np.float64: 1e-13}
+      atol = {np.float32: 6e-2, np.float64: 1e-13}
+    else:
+      rtol = {np.float32: 2e-5, np.float64: 1e-13}
+      atol = {np.float32: 5e-5, np.float64: 1e-13}
+
     if jit_f:
       f = jax.jit(f)
     if jit_scan:
@@ -1672,15 +1682,6 @@ class LaxControlFlowTest(jtu.JaxTestCase):
 
     ans = jax.grad(lambda c, as_:      list(          scan(f, c, as_))[0].sum())(c, as_)
     expected = jax.grad(lambda c, as_: list(scan_reference(f, c, as_))[0].sum())(c, as_)
-    if scan is scan_with_new_checkpoint:
-      rtol = {np.float32: 5e-5, np.float64: 1e-13}
-      atol = 1e-5
-    elif scan is scan_with_for:
-      rtol = {np.float32: 2e-5, np.float64: 1e-13}
-      atol = {np.float32: 6e-2, np.float64: 1e-13}
-    else:
-      rtol = {np.float32: 2e-5, np.float64: 1e-13}
-      atol = 1e-5
     self.assertAllClose(ans, expected, check_dtypes=False, rtol=rtol, atol=atol)
 
     rtol = 5e-3 if scan is not scan_with_new_checkpoint2 else 5e-2

